makefile: add missing TARGET for android-clean

jakubgs opened the pull request:
Changes:
* Fixed `make android-clean` target failing due to lack of Gradle
* Included small change that moves `make nix-purge` to `nix/scripts/purge.sh` for simplicity
* Fixed Haste warnings about mode name collisions by changing blacklist in `rn-cli.config.js`
* Got rid of the old `TARGET_OS` variable from `Makefile`

jakubgs committed:
makefile: add missing TARGET for android-clean

Signed-off-by: Jakub Sokołowski <jakub@status.im>

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 2edb8014 | [#1](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-10376/1/) | 2020-04-21 10:25:15 | ~8 min | `ios` | [:package:`ipa`](https://i.diawi.com/XUg1ZN) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2FXUg1ZN)|

jakubgs committed:
nix: move nix-purge code to nix/scripts/purge.sh

Signed-off-by: Jakub Sokołowski <jakub@status.im>

jakubgs committed:
nix: completely remove legacy TARGET_OS variable

Signed-off-by: Jakub Sokołowski <jakub@status.im>

jakubgs committed:
react-native: avoid Haste module naming collision warnings

Warnings like this:

jest-haste-map: Haste module naming collision: StatusIm-Mobile
  The following files share their name; please adjust your hasteImpl:
    * <rootDir>/mobile/js_files/package.json
    * <rootDir>/package.json

Signed-off-by: Jakub Sokołowski <jakub@status.im>
